tests/monitoring_processes/test_monitoring_critical_processes.py

```diff
@@ -0,0 +1,447 @@
+"""
+Test the feature of monitoring critical processes by Supervisord.
+"""
+from collections import defaultdict
+import logging
+
+import pytest
+
+from pkg_resources import parse_version
+from tests.common import config_reload
+from tests.common.helpers.assertions import pytest_assert
+from tests.common.helpers.dut_utils import get_program_info
+from tests.common.helpers.dut_utils import get_group_program_info
+from tests.common.plugins.loganalyzer.loganalyzer import LogAnalyzer, LogAnalyzerError
+from tests.common.utilities import wait_until
+
+logger = logging.getLogger(__name__)
+
+pytestmark = [
+    pytest.mark.topology('any'),
+    pytest.mark.disable_loganalyzer
+]
+
+CONTAINER_CHECK_INTERVAL_SECS = 1
+CONTAINER_RESTART_THRESHOLD_SECS = 180
+
+
+@pytest.fixture(autouse=True, scope='module')
+def config_reload_after_tests(duthost):
+    yield
+    config_reload(duthost)
+
+
+@pytest.fixture(autouse=True, scope='module')
+def disable_and_enable_autorestart(duthost):
+    """Changes the autorestart of containers from `enabled` to `disabled` before testing.
+       and Rolls them back after testing.
+
+    Args:
+        duthost: Hostname of DUT.
+
+    Returns:
+        None.
+    """
+    containers_autorestart_states = duthost.get_container_autorestart_states()
+    disabled_autorestart_containers = []
+
+    for container_name, state in containers_autorestart_states.items():
+        if state != "disabled":
```

Comment:
Can you disable always_enabled feature?

Reply:
`always_enable` state is immutable and can not be changed.